Sources/App/WebView/WebViewWindowController.swift

```diff
@@ -202,9 +202,31 @@ class WebViewWindowController {
         }
     }
 
-    func openSelectingServer(from: OpenSource, urlString openUrlRaw: String, skipConfirm: Bool = false) {
-        if let first = Current.servers.all.first, Current.servers.all.count == 1 {
-            open(from: from, server: first, urlString: openUrlRaw, skipConfirm: skipConfirm)
+    func openSelectingServer(
+        from: OpenSource,
+        urlString openUrlRaw: String,
+        skipConfirm: Bool = false,
+        queryParameters: [URLQueryItem]? = nil
+    ) {
+        let serverName = queryParameters?.first(where: { $0.name == "server" })?.value
+        let servers = Current.servers.all
+
+        if let first = servers.first, Current.servers.all.count == 1 || serverName != nil {
+            if serverName == "default" || serverName == nil {
+                open(from: from, server: first, urlString: openUrlRaw, skipConfirm: skipConfirm)
+            } else {
+                if let selectedServer = servers.first(where: { server in
+                    let sanitizedServerName = server.info.name.lowercased()
+                        .addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)
```

Comment:
doesn't the URLQueryItem version of this strip url encoding?

Reply:
I don't remember why I did that but I remember I tested some scenarios that the browser allowed but then the comparison would fail... something related to non ASCII chars.
I'll test again and get back to you

Reply:
Good catch, I have updated the PR

Reply:
if the value in the URLQueryItem is decoded already, wouldn't this make something 'weird' not match?

Reply:
yeah so this introduces an issue with spaces here. `?server=dev%20local` looks like:

```
sanitizedServerName: "dev%20local"
serverName: "dev local"
```

i dont' think we need to do any url in this line - we're on both sides of the "not encoded"

Reply:
You're correct "addingPercentEncoding" was irrelevant there, I have removed it